modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/fragment/renderer/OrdersDataSetFragmentRenderer.java

```diff
@@ -429,6 +431,26 @@ private CreationMenu _getFDSCreationMenu(
 		return null;
 	}
 
+	private String _getOrderURL(HttpServletRequest httpServletRequest)
```

Comment:
Just a question: I see here we're introducing cross-compatibility between the Orders Data Set fragment and the Open Order portlet, which is fine IMO. But this also says that, whenever the portlet is in place, we're always prioritizing it over the new order pages, correct?

Is this what we want? Wouldn't it be better to find a way in order to check:
```
if => there's an Order friendly URL route
else if => there's the Open Order portlet in the site
else => fail
```

The first case will be most likely always false for the old Minium site init and will fallback to the second case.
But it'll be most likely true for the new Classic-based site init's but still will behave correctly if there's the old Open Order Portlet and you don't want to use the order pages.

Wdyt?

Reply:
I agree with this proposal, it looks more compliant with the requirements we had for order pages

Reply:
The first if statement will always be true, because the friendlyURL has always a value. The issue is that we can check if an order has a DPT only once it is created and in a Java context. In this case I just reused the MVCActionCommand, but the best would be to have a new non-headless rest application to create the order and understand where to redirect the user.